dags/hindawi/hindawi_api_client.py

```diff
@@ -2,25 +2,29 @@
 from xml.etree import ElementTree
 
 from common.request import Request
+from requests.exceptions import RequestException
 
 
 class HindawiApiClient:
     def __init__(
         self,
         base_url=None,
-        path_segments=None,
+        files_url=None,
     ):
         self.base_url = base_url or getenv(
             "HINDAWI_API_BASE_URL", "https://www.hindawi.com"
         )
-        self.path_segments = path_segments or ["oai-pmh", "oai.aspx"]
+        self.files_url = files_url or getenv(
+            "HINDAWI_API_FILES_URL", "http://downloads.hindawi.com"
+        )
 
     def get_articles_metadata(self, parameters, doi=None):
+        path_segments = ["oai-pmh", "oai.aspx"]
         if doi:
-            self.path_segments.append(doi)
+            path_segments.append(doi)
```

Comment:
do we have a cease with an article without doi? If the doi is None, what the request will return?

Reply:
All articles by the date, not just one

Reply:
so all the articles with dates don't have a doi?

Reply:
We don't need, because we can make a request by:
* range of dates, we will get all articles published in this date range
* doi, just one specific article